catalog/transformer/catalog-transformer-streaming-lib/src/main/java/org/codice/ddf/transformer/xml/streaming/lib/XmlInputTransformer.java

```diff
@@ -260,9 +269,37 @@ public MetacardType getMetacardType() {
             }
         }
         attributeDescriptors.addAll(BasicTypes.BASIC_METACARD.getAttributeDescriptors());
-        return new DynamicMetacardType(attributeDescriptors, id);
+
+        DynamicMetacardType dynamicMetacardType = new DynamicMetacardType(attributeDescriptors, id);
+        registerMetacardType(dynamicMetacardType);
+        return dynamicMetacardType;
+    }
+
+    private void registerMetacardType(MetacardType metacardType) {
+        unregisterMetacardType(metacardType);
+
+        ServiceRegistration serviceRegistration =
+                context.registerService(MetacardType.class.getName(), metacardType, null);
```

Comment:
You should include service properties that has the name of the metacard.
https://github.com/codice/ddf/blob/master/catalog/transformer/catalog-transformer-pdf/src/main/resources/OSGI-INF/blueprint/blueprint.xml#L33